src/util/locking/FileSystemResourceLocker.ts

```diff
@@ -0,0 +1,146 @@
+import { createHash } from 'crypto';
+import { ensureDirSync, pathExists, readdir, rmdir } from 'fs-extra';
+import type { LockOptions, UnlockOptions } from 'proper-lockfile';
+import { lock, unlock } from 'proper-lockfile';
+import type { ResourceIdentifier } from '../../http/representation/ResourceIdentifier';
+import type { Finalizable } from '../../init/final/Finalizable';
+import { getLoggerFor } from '../../logging/LogUtil';
+import { createErrorMessage } from '../errors/ErrorUtil';
+import { InternalServerError } from '../errors/InternalServerError';
+import type { AttemptSettings } from '../LockUtils';
+import { toRetryOnError, retryFunctionUntil } from '../LockUtils';
+import { joinFilePath } from '../PathUtil';
+import type { ResourceLocker } from './ResourceLocker';
+
+const defaultLockOptions: LockOptions = {
+  /** Resolve symlinks using realpath, defaults to true (note that if true, the file must exist previously) */
+  realpath: false,
+  /** The number of retries or a [retry](https://www.npmjs.org/package/retry) options object, defaults to 0 */
```

Comment:
Do markdown links work in TSDoc? Genuine question, never thought about it 😄

Reply:
I assumed, since visual code seems to render them on the documentation of functions.